src/scripts/costing/cost_estimation.py

```diff
@@ -0,0 +1,2280 @@
+        (cost_inputs['actual_expenditure_data']['Cost Type'].notna()) &
+        (cost_inputs['actual_expenditure_data']['Cost Type'] != "Grand Total")]
+
+    first_nan_index_facility_operations = cost_inputs['facility_operations'][
+        cost_inputs['facility_operations']['Facility_Level'].isna()].index.min()
+    if pd.notna(first_nan_index_facility_operations):
+        cost_inputs['facility_operations'] = cost_inputs['facility_operations'].loc[
+                                             :first_nan_index_facility_operations - 1]
+
+    cost_inputs['health_spending_projections'].columns = cost_inputs['health_spending_projections'].iloc[1]
+    cost_inputs['health_spending_projections'] = cost_inputs['health_spending_projections'].iloc[2:].reset_index(
+        drop=True)  # Assign the fourth row as column names
+
+    first_nan_index_health_spending = cost_inputs['health_spending_projections'][
+        cost_inputs['health_spending_projections']['year'].isna()].index.min()
+    if pd.notna(first_nan_index_health_spending):
+        cost_inputs['health_spending_projections'] = cost_inputs['health_spending_projections'].loc[
+                                                     :first_nan_index_health_spending - 1]
+
+    cost_inputs['health_spending_projections']['year'] = cost_inputs['health_spending_projections']['year'].astype(int)
+
+    return cost_inputs
+
+
+# Define a helper function to load necessary simulation metadata
+def load_simulation_metadata(results_folder: Path) -> tuple[dict, list[int], tuple[Date, Date]]:
+    """
+    Load simulation scenario metadata and derive key parameters.
+
+    Parameters
+    ----------
+    results_folder : Path
+        Path to the folder containing TLO model simulation results.
+
+    Returns
+    -------
+    info : dict
+        Scenario metadata including number of draws and runs per draw.
+    years : list of int
+        Full list of simulation years.
+    target_period : tuple of Date
+        Simulation date range from first to last year as TLO Date objects.
+    """
+    # Load a sample log to derive time bounds
+    log = load_pickled_dataframes(results_folder, 0, 0)
+    dates = log['tlo.methods.healthsystem.summary']['hsi_event_counts']['date']
+    first_year = min(dates).year
+    last_year = max(dates).year
+    years = list(range(first_year, last_year + 1))
+    target_period = (Date(first_year, 1, 1), Date(last_year, 12, 31))
+
+    # Get simulation info (number of draws, runs, etc.)
+    info = get_scenario_info(results_folder)
+
+    return info, years, target_period
+
+
+# Define a function to discount and summarise costs by cost_category
+def apply_discounting_to_cost_data(_df: pd.DataFrame,
+                                   _discount_rate: Union[float, dict[int, float]] = 0,
+                                   _initial_year: Optional[int] = None,
+                                   _column_for_discounting: str = 'cost') -> pd.DataFrame:
```

Comment:
here, and elsewhere, arguments are named with the leading underscore.

Whilst this all works perfectly, the convention is for "_" to be used for private members, and so it's unusual to have them as argument names. 

https://betterstack.com/community/questions/what-does-underscores-before-object-name-mean-in-python/

So, to improve future readability, I would propose a refactor to just rename the arguments to be without the leading underscore.